apps/core/scenes/results/ResultsListItem.js

```diff
@@ -1,61 +1,195 @@
 // @flow strict
 
 import * as React from 'react';
+import { View } from 'react-native';
+import * as DateFNS from 'date-fns';
 import { graphql, createFragmentContainer } from '@kiwicom/margarita-relay';
-import { Text, StyleSheet } from '@kiwicom/universal-components';
+import { StyleSheet } from '@kiwicom/universal-components';
 import { defaultTokens } from '@kiwicom/orbit-design-tokens';
+import {
+  ConnectionCard,
+  type TripSectorWithId,
+} from '@kiwicom/margarita-components';
 
 import type { ResultsListItem as ResultsListItemType } from './__generated__/ResultsListItem.graphql';
 
 type Props = {|
   +data: ResultsListItemType,
 |};
 
-const ResultListItem = (props: Props) => {
-  if (props.data == null) {
+export type Route = $PropertyType<ResultsListItemType, 'route'>;
+export type RouteItem = $ElementType<$NonMaybeType<Route>, number>; // number because arrays are number-indexed
+
+// @TODO get from config
+const dateFormat = 'ddd D MMM';
+
+class ResultListItem extends React.Component<Props> {
+  parseDate = (date: ?Date) => {
+    if (date) return DateFNS.parse(date);
     return null;
-  }
-  const { flyFrom, flyTo, localDeparture, localArrival, price } = props.data;
-  /**
-   * TODO: properly handle possible undefined props in final version of list
-   * (for example: should be entry with undefined price still displayed?)
-   *
-   * NOTE: ugly form of string creation below is used for now because
-   * using ?? inside template literals caused problems with code styling
-   * inside VS Code (this part will be probably completely replaced
-   * in next iteration, so it's temporary solution)
-   */
-  const itemString =
-    (flyFrom ?? 'Unknown destination') +
-    ' / ' +
-    (localDeparture ?? 'Unknown departure') +
-    ' -> ' +
-    (flyTo ?? 'Unknown destination') +
-    ' / ' +
-    (localArrival ?? 'Unknown arrival') +
-    ' | ' +
-    (price ?? 'Unknown price');
-  return <Text style={styles.listItem}>{itemString}</Text>;
-};
+  };
 
-const styles = StyleSheet.create({
-  listItem: {
-    fontSize: 12,
-    padding: 2,
-    margin: 2,
-    backgroundColor: defaultTokens.backgroundBody,
-  },
-});
+  // @TODO Handle cases when we don't have some data from GraphQL/REST server
+  getSanitizedRouteItem = (route: RouteItem) => {
+    const timeSimpleFormat = 'H:mm';
+
+    const localArrival = this.parseDate(route?.arrival?.localTime);
+    const localDeparture = this.parseDate(route?.departure?.localTime);
+    const utcArrival = this.parseDate(route?.arrival?.utcTime);
+    const utcDeparture = this.parseDate(route?.departure?.utcTime);
+    const duration =
+      route &&
+      `${DateFNS.differenceInHours(
+        utcArrival,
+        utcDeparture,
+      )}h ${DateFNS.differenceInMinutes(utcArrival, utcDeparture) % 60}m`;
+
+    return {
+      arrival: (route && route.arrival?.city) ?? '',
```

Comment:
It looks like you're already sanitising your GraphQL server outputs, so maybe this is not necessary now.
My point before was that a GraphQL schema should probably contain nullable types, instead of `GraphqlString`, etc. in which case sanitising in the client makes sense. 
@FilipMessa @tbergq What are your thoughts about nullable types by default in GraphQL schemas?

Reply:
It should be nullable, It can be annoying handle all this nullable stuff, but on the end, it pushes you as a developer thinking about this possibility. And, yes we are working here with the graphql as a proxy, but we still can rely on that all data will come. 
We should analyze what data are necessary to handle Bisnis logick on the FE, and base on that decided if we show rest of data or some error message. But it needs to analyze what data are important, so, for now, I suggest use "N/A" constant instead of empty string.

What do you think about that?

Reply:
Sounds good. Done